integration-tests/src/tests/client/chunks_management.rs

```diff
@@ -297,7 +326,74 @@ fn chunks_recovered_from_full() {
         run_actix(async {
             chunks_produced_and_distributed_common(
                 4,
+                false,
+                &["test1"],
+                2 * CHUNK_REQUEST_SWITCH_TO_FULL_FETCH_MS,
+            );
+        })
+    });
+}
+
+// And now let's add chunk-only producers into the mix
+
+/// Happy case -- each shard is handled by one cop and one block producers.
+#[test]
+#[cfg(feature = "protocol_feature_chunk_only_producers")]
+fn chunks_produced_and_distributed_one_val_shard_cop() {
+    heavy_test(|| {
+        run_actix(async {
+            chunks_produced_and_distributed_common(4, true, &[], 15 * CHUNK_REQUEST_RETRY_MS);
+        });
+    });
+}
+
+/// `test4` can't talk to `test1`, so it'll fetch the chunk for first shard from `cop1`.
+#[test]
+#[cfg(feature = "protocol_feature_chunk_only_producers")]
+fn chunks_recovered_from_others_cop() {
+    heavy_test(|| {
+        run_actix(async {
+            chunks_produced_and_distributed_common(
+                1,
                 true,
+                &["test1"],
+                4 * CHUNK_REQUEST_SWITCH_TO_OTHERS_MS,
+            );
+        });
+    });
+}
+
+/// `test4` can't talk neither to `cop1` nor to `test1`, so it can't fetch chunk
+/// from chunk producers and has to reconstruct it.
+///
+/// I would expect this to fail for the same reason
+/// `chunks_recovered_from_full_timeout_too_short` fails, but somehow it works?
+#[test]
+#[should_panic]
+#[cfg(feature = "protocol_feature_chunk_only_producers")]
+fn chunks_recovered_from_full_timeout_too_short_cop() {
+    heavy_test(|| {
+        run_actix(async {
+            chunks_produced_and_distributed_common(
+                4,
+                true,
+                &["test1", "cop1"],
+                4 * CHUNK_REQUEST_SWITCH_TO_OTHERS_MS,
+            );
+        });
+    });
+}
```

Comment:
@mzhangmzz so this one baffles me: `chunks_recovered_from_full_timeout_too_short` (without cops) is a `should_panic` test we expect it to fail.

But somehow, if I add cops, it actually passes. 

I don't have a great understanding of what *actually* should be happening here.